source/OptimaJet.DWKit.StarterApplication/Forms/Projects/CreateForm.cs

```diff
@@ -0,0 +1,94 @@
+            if ((project.OrganizationId != 0)
+                && (project.OwnerId != 0)
+                && (project.GroupId != 0))
+            {
+                var organization = OrganizationRepository.Get()
+                        .Where(o => o.Id == project.OrganizationId)
+                        .FirstOrDefaultAsync().Result;
+                var group = GroupRepository.Get()
+                       .Where(g => g.Id == project.GroupId)
+                       .Include(g => g.Owner).FirstOrDefaultAsync().Result;
+                var owner = UserRepository.Get()
+                        .Where(u => u.Id == project.OwnerId)
+                        .Include(u => u.GroupMemberships)
+                            .ThenInclude(gm => gm.Group)
+                        .Include(u => u.OrganizationMemberships)
+                            .ThenInclude(om => om.Organization)
+                        .FirstOrDefaultAsync().Result;
+
+                if ((organization == null) || (group == null) || (owner == null))
+                {
+                    // Allowing it to return in these cases should cause the base to hit normal
+                    // foreign key failures
+                    var message = $"Project '{project.Name}': Organization Group or Owner not present or invalid";
+                    Log.Error(message);
+                    return true;
+                }
+                if (organization != group.Owner)
+                {
+                    retVal = false;
+                    var message = $"Project '{project.Name}': Group '{group.Name}' not owned by project organization '{organization.Name}'";
+                    Log.Error(message);
+                    var error = new Error(422, "Project Validation Failure", message);
+                    Errors.Add(error);
+
+                }
+                if ((owner.OrganizationIds == null) || (!owner.OrganizationIds.Contains(organization.Id)))
+                {
+                    retVal = false;
+                    var message = $"Project '{project.Name}': Owner '{owner.Name}' not a member of project organization '{organization.Name}'";
+                    var error = new Error(422, "Project Validation Failure", message);
+                    Errors.Add(error);
+                }
+                // The current user should be a member of the organization
+                var currentUser = UserRepository.GetByAuth0Id(CurrentUserContext.Auth0Id).Result;
+                var orgIds = currentUser.OrganizationIds ?? Enumerable.Empty<int>();
+                if (!orgIds.Contains(organization.Id))
+                {
+                    retVal = false;
+                    var message = $"Project '{project.Name}': Current user not a member of project organization '{organization.Name}'";
+                    var error = new Error(422, "Project Validation Failure", message);
+                    Errors.Add(error);
+                }
+ 
+            }
+            return retVal;
```

Comment:
same not here about checking the errors object.
maybe having the superclass do the comparison, so at the end of each of these, you can just do `base.IsValid()`

Reply:
Fixed as above